### PR TITLE
Update ledger soft batch rpc tests

### LINKED ISSUE #320: Update Citrea RPC tests
RPC tests are not updated and right now are based on the original sovereign-sdk structure.

We should test RPCs we added.

- [ ] ledger_getSoftBatchRange
- [ ] ledger_getSoftBatchByNumber
- [ ] ledger_getSoftBatchByHash
- [ ] ledger_getSoftConfirmationStatus

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Updates ledger rpc soft batch tests

## Linked Issues
- Fixes #320

### CODE REVIEW

File: crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs [hunk 1]
```diff
@@ -59,9 +61,9 @@ impl LedgerRpcProvider for LedgerDB {
 
     fn get_soft_batch(
         &self,
-        batch_id: SoftBatchIdentifier,
+        batch_id: &SoftBatchIdentifier,
     ) -> Result<Option<SoftBatchResponse>, anyhow::Error> {
-        let batch_num = self.resolve_soft_batch_identifier(&batch_id)?;
+        let batch_num = self.resolve_soft_batch_identifier(batch_id)?;
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs:66]: what does self.resolve_soft_batch_identifier return?

if it is something line Num/Hash

lets update this function

File: crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs [hunk 2]
```diff
@@ -286,6 +288,43 @@ impl LedgerRpcProvider for LedgerDB {
...
+        Ok(out)
+    }
+
+    fn get_soft_batches_range(
+        &self,
+        start: u64,
+        end: u64,
+    ) -> Result<Vec<Option<SoftBatchResponse>>, anyhow::Error> {
+        anyhow::ensure!(start <= end, "start must be <= end");
+        anyhow::ensure!(
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs:319]: math's wrong

start 10
end 30 
max allowed is 20

i get to take 21 soft batches lol

File: crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs [hunk 3]
```diff
@@ -347,7 +386,10 @@ impl LedgerDB {
         batch_id: &SoftBatchIdentifier,
     ) -> Result<Option<BatchNumber>, anyhow::Error> {
         match batch_id {
-            SoftBatchIdentifier::Hash(hash) => self.db.get::<SoftBatchByHash>(hash),
+            SoftBatchIdentifier::Hash(hash) => {
+                println!("Hash: {:?}", hash);
```
Comment (eyusufatik) [inline crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs:390]: 🤔
no unnecessary prints pls

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs (hunk 1 above)
Comment (ercecan) [reply crates/sovereign-sdk/full-node/db/sov-db/src/ledger_db/rpc.rs:66]: returns batch_num

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**